### ISSUE TITLE
setState Can only update a mounted or mounting component

### ISSUE THREAD

**Issue by QuietOmen** [leandrowd/react-responsive-carousel#30]
I get the following errors: 

```
Warning: setState(...): Can only update a mounted or mounting component. This usually means you called setState() on an unmounted component. This is a no-op. Please check the code for the Slider component.

Uncaught TypeError: Cannot read property 'clientWidth' of null
```
It seems to stem back to the following function inside of Carousel.js

```
setMountState () {
     this.setState({hasMount: true});
     this.updateSizes();
     this.forceUpdate();
},
```

The clientWidth of null comes from the updateSizes function being called. I don't believe we are incorrectly using the library. Here's where we use the carousel component: 

```
import React, {PropTypes} from 'react';
import {Carousel} from 'react-responsive-carousel';
import './styles';

export function CarouselComponent({data}) {
  if (!data || data['total-images'] === 0) {
    return <span />;
  }
  return (
    <div className="carousel-component">
      <Carousel
        showThumbs={data.gallery.length > 1}
        showArrows={true}
        showStatus={false}
        showIndicators={false}
      >
        {
          data.gallery.map(
            (item, index) => <img src={item['image-url']} key={index} />
          )
        }
      </Carousel>
      <div className="photo-count">
        <span className="swipe-to-see">
          Swipe to see&nbsp;
        </span>
        {data['total-images']} photos
      </div>
    </div>
  );
}

CarouselComponent.propTypes = {
  data: PropTypes.any,
};
```

Also we're on react-responsive-carousel 3.0.9.

**Comment by leandrowd** [comment]
Hi @QuietOmen, thanks for raising the issue.

Seems that you are rendering a carousel without any items inside (data.gallery = []). Is that the case? 

I won't be able to do any fix to this plugin right now but if you keen on changing it, please submit a Pull Request and I will merge it and publish to npm. Otherwise, I can fix when I have time.

Thanks.

**Comment by leandrowd** [comment]
Related to #29

**Comment by leandrowd** [comment]
This issue was fixed and released as 3.0.16
